builtin/logical/mysql/secret_creds.go

```diff
@@ -59,27 +67,58 @@ func (b *backend) secretCredsRevoke(
 		return nil, err
 	}
 
+	// Get the role name
+	rolenameRaw, ok := req.Secret.InternalData["rolename"]
+	if !ok {
+		return nil, fmt.Errorf("secret is missing rollname internal data")
+	}
+	rolename, ok := rolenameRaw.(string)
+
+	role, err := b.Role(req.Storage, rolename)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return logical.ErrorResponse(fmt.Sprintf("unknown role: %s", rolename)), nil
+	}
+
 	// Start a transaction
 	tx, err := db.Begin()
 	if err != nil {
 		return nil, err
 	}
 	defer tx.Rollback()
 
-	// Revoke all permissions for the user. This is done before the
-	// drop, because MySQL explicitly documents that open user connections
-	// will not be closed. By revoking all grants, at least we ensure
-	// that the open connection is useless.
-	_, err = tx.Exec("REVOKE ALL PRIVILEGES, GRANT OPTION FROM '" + username + "'@'%'")
-	if err != nil {
-		return nil, err
+	// Check for an empty revokeSQL string
+	// set it to a default query if the string is empty
+	if role.RevokeSQL == "" {
+		role.RevokeSQL = "REVOKE ALL PRIVILEGES, GRANT OPTION FROM '" + username + "'@'%'; DROP USER '" + username + "'@'%'"
```

Comment:
Make this a `const` in the file rather than defining it here.

More generally your logic needs to change so that this is used if either there is no role (the role name was empty) or the role has no defined revocation sql.